### PR TITLE
Add support for managing key pair resources

### PR DESCRIPTION
**Pull request by rmoe**
This implements the CRUD operations for the key pair
resource. Importing is not supported as there's no way to retrieve the
private key from travis after creation.

### CODE REVIEW

**Comment by rmoe** [comment]
@bgpat have you had a chance to look at this?

**Comment by bgpat** [comment]
Thank you your contribution! I'll review this weekend.

File: travis/resource_key_pair.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+	if d.HasChange("value") {
+		update.Value = d.Get("value").(string)
+	}
+
+	if d.HasChange("description") {
+		update.Description = d.Get("description").(string)
+	}
+
+	if update.Value != "" || update.Description != "" {
```
Comment (bgpat) [inline travis/resource_key_pair.go:128]: I seem Travis accepts an empty description.

```suggestion
	if update.Value != "" {
```

File: travis/resource_key_pair_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+		result, _, err := client.KeyPair.FindByRepoSlug(context.Background(), testRepoSlug)
+		if err != nil {
+			return err
+		}
+		*keyPair = *result
+		return nil
+	}
+}
+
+func testAccKeyPairResource(desc string, testAccPrivateKey string) string {
```
Comment (bgpat) [inline travis/resource_key_pair_test.go:81]:
```suggestion
func testAccKeyPairResource(desc, testAccPrivateKey string) string {
```

**Review by bgpat — CHANGES_REQUESTED** [review]
Run `go generate ./...` to generate docs.

File: travis/resource_key_pair.go (hunk 1 above)
Comment (rmoe) [reply travis/resource_key_pair.go:128]: Wouldn't this mean that the description would only be updated if you also updated the value at the same time?

I did a little testing and the Travis API does accept an empty description on both creation and update. The client library here won't update the description to an empty string from what I can tell.

```go
package main

import (
	"context"
        "fmt"
	"github.com/shuheiktgw/go-travis"
)

func main() {
	client := travis.NewClient(...)
	repoSlug := "repo slug"

	keyPair, _, _ := client.KeyPair.FindByRepoSlug(context.Background(), repoSlug)
	fmt.Printf("Existing keypair description: %s\n", *keyPair.Description)

	update := &travis.KeyPairBody{
		Description: "",
	}
	keyPair, _, _ = client.KeyPair.UpdateByRepoSlug(context.Background(), repoSlug, update)
	fmt.Printf("Keypair description returned by PATCH: %s\n", *keyPair.Description)

	keyPair, _, _ = client.KeyPair.FindByRepoSlug(context.Background(), repoSlug)
	fmt.Printf("Keypair desc fetched again: %s\n", *keyPair.Description)
}
```
If I run this against one of my repos I get this:

```
Existing keypair description: Service Key
Keypair description returned by PATCH: Service Key
Keypair desc fetched again: Service Key
```

I was able to use curl to set an empty description on that same keypair however.

**Review by bgpat — APPROVED** [review]
LGTM

**Pull request merged**